package/src/index/scrape.rs

```diff
@@ -33,23 +35,51 @@ pub fn fetch_git(id: &Id, commit: &ObjectId) -> Result<Package, Error> {
     let tmpdir = tempdir().without_path()?;
     let _id = nickel_lang_git::fetch(&Spec::commit(id.remote_url()?, *commit), tmpdir.path())?;
 
-    let manifest_path = tmpdir.path().join(MANIFEST_NAME);
+    let Id::Github { path, org, name } = id.clone();
+
+    let mut manifest_path = tmpdir.path().to_owned();
+    manifest_path.push(&path);
```

Comment:
What happens if `path` is empty here? I couldn't find an answer in the documentation of `PathBuf::push`, so I don't think we should rely on any assumption here (like pushing an empty string is a no-op). In general, I'm feeling slightly uncomfortable about the `path` not being and `Option` (even if I get that we use an empty string as an `None` value, but semantically, it's a bit strange to have a Nickel schema with `| optional` but an always-defined value on the Rust side).

Reply:
I added a check here to avoid pushing an empty path. I'm not sure about the `Option`, though. I agree it matches the contract better, but it also provides two different representations of the same thing (`None` and `Some("")`), which makes it possible to accidentally have different behavior.

I added a comment to the place where undefined becomes empty. Does that help?

Reply:
I suppose the best option on paper would be something like `Option<NonEmptyString>`, but I agree that `String` is an ok approximation. Should be good with the explicit comment :+1: